test/controllers/RequestHandlerControllerSpec.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.BaseSpec
+import mocks.MockDataRepository
+import models.DynamicDataModel
+import play.api.libs.json.Json
+import play.mvc.Http.Status
+
+class RequestHandlerControllerSpec extends BaseSpec with MockDataRepository {
+
+  object TestRequestHandlerController extends RequestHandlerController(mockDataRepository)
```

Comment:
Do we need an object here? It doesn't really extend the functionality of RequestHandlerController. Can you create a new RequestHandlerController instead?

Reply:
This is creating a new `RequestHandlerController`. It's just doing it in a different way than assigning a `val` and using the `new` keyword. I do prefer the `val` way of doing it though as I think it is more clear, so I'll change it.